controllers/nstemplateset/cluster_resources.go

```diff
@@ -327,12 +328,35 @@ func (r *clusterResourcesManager) delete(logger logr.Logger, nsTmplSet *toolchai
 
 // isUpToDate returns true if the currentObject uses the corresponding templateRef and tier labels
 func isUpToDate(currentObject, _ runtimeclient.Object, tierTemplate *tierTemplate) bool {
+	// --- start temporary logic
+	// this will trigger an update in order to set the SpaceLabelKey on all the objects with the OwnerLabelKey
+	if !hasSpaceLabelSet(currentObject) {
+		return false
+	}
+	// -- end of temporary migration logic
+
 	return currentObject.GetLabels() != nil &&
 		currentObject.GetLabels()[toolchainv1alpha1.TemplateRefLabelKey] == tierTemplate.templateRef &&
 		currentObject.GetLabels()[toolchainv1alpha1.TierLabelKey] == tierTemplate.tierName
 	// && currentObject.IsSame(newObject)  <-- TODO Uncomment when IsSame is implemented for all ToolchainObjects!
 }
 
+func hasSpaceLabelSet(currentObject runtimeclient.Object) bool {
+	if currentObject.GetLabels() == nil {
+		return false
+	}
+
+	ownerValue, hasOwnerLabelKey := currentObject.GetLabels()[toolchainv1alpha1.OwnerLabelKey]
+	if hasOwnerLabelKey {
```

Comment:
Yeah, that true, but I didn't want to completely rely/trust on the "client", I thought that it may be more safe and clear checking it in the function.